src/Abc.Zebus.MessageDsl/Analysis/AstValidator.cs

```diff
@@ -99,21 +99,31 @@ private void DetectDuplicateTypes()
 
             foreach (var typeNode in types)
             {
-                var name = ((INamedNode)typeNode).Name;
+                var id = Id(typeNode);
 
-                if (!seenTypes.Add(name))
-                    duplicates.Add(name);
+                if (!seenTypes.Add(id))
+                    duplicates.Add(id);
             }
 
             foreach (var typeNode in types)
             {
-                var name = ((INamedNode)typeNode).Name;
+                var id = Id(typeNode);
 
-                if (duplicates.Contains(name))
-                    _contracts.AddError(typeNode.ParseContext, "Duplicate type name: {0}", name);
+                if (duplicates.Contains(id))
+                    _contracts.AddError(typeNode.ParseContext, "Duplicate type id: {0}", id);
             }
         }
 
+        private string Id(AstNode node)
```

Comment:
Rename the method to something like `GetNameWithGenericArity`, and make this an inner method of `DetectDuplicateTypes`